### PR TITLE
[SPARK-35948][INFRA] Simplify release scripts by removing Spark 2.4/Java7 parts

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to clean up Spark 2.4 and Java7 code path from the release scripts.

### Why are the changes needed?

To simplify the logic.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @gengliangwang , @srowen , @cloud-fan , @HyukjinKwon

File: dev/create-release/release-util.sh [hunk 1]
```diff
@@ -226,11 +226,5 @@ function init_maven_sbt {
   MVN="build/mvn -B"
   MVN_EXTRA_OPTS=
   SBT_OPTS=
-  if [[ $JAVA_VERSION < "1.8." ]]; then
-    # Needed for maven central when using Java 7.
-    SBT_OPTS="-Dhttps.protocols=TLSv1.1,TLSv1.2"
-    MVN_EXTRA_OPTS="-Xmx2g -XX:ReservedCodeCacheSize=1g -Dhttps.protocols=TLSv1.1,TLSv1.2"
-    MVN="$MVN $MVN_EXTRA_OPTS"
-  fi
```
Comment (dongjoon-hyun) [inline dev/create-release/release-util.sh:234]: It's okay to fail for Java 7. So, we don't need this.

**Review by gengliangwang — APPROVED** [review]
LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you, @gengliangwang and @HyukjinKwon !

**Pull request closed without merging**